doc/03_Documents/03_Navigation.md

```diff
@@ -356,6 +356,70 @@ class NavigationExtension extends AbstractExtension
 </div>
 ```
 
+## Adding Custom Items to the Root Navigation
+
+In the following example we're adding news categories (objects) to the root navigation (1st level).
+```php
+<?php
+
+namespace App\Twig\Extension;
+
+use App\Website\LinkGenerator\NewsCategoryLinkGenerator;
+use Pimcore\Model\Document;
+use Pimcore\Navigation\Container;
+use Pimcore\Twig\Extension\Templating\Navigation;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class NavigationExtension extends AbstractExtension
+{
+    protected Navigation $navigationHelper;
+    protected NewsCategoryLinkGenerator $newsCategoryLinkGenerator;
+
+    public function __construct(Navigation $navigationHelper, NewsCategoryLinkGenerator $newsCategoryLinkGenerator)
+    {
+        $this->navigationHelper = $navigationHelper;
+        $this->newsCategoryLinkGenerator = $newsCategoryLinkGenerator;
+    }
+
+    /**
+     * @return TwigFunction[]
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('app_navigation_news_categories', [$this, 'newsCategoryNavigation'])
+        ];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function newsCategoryNavigation(Document $document, Document $startNode): Container
+    {
+        $navigation = $this->navigationHelper->build([
+            'active' => $document,
+            'root' => $startNode,
+            'rootCallback' => function(Container $navigation) {
+                $list = new \Pimcore\Model\DataObject\NewsCategory\Listing;
+                $list->load();
+                foreach($list as $category) {
+                    $detailLink = $this->newsCategoryLinkGenerator->generate($category);
```

Comment:
Thank you for the requested changes and the typo fix in the docs!

I would stick to a plainer example with a more hardcoded label and url, otherwise we may need to provide `newsCategoryLinkGenerator` somewhere too, eg. in the demo repository or a snippet here in the docs.

Also would use the example for `pageCallback` and merge the 2, to show it can be both used on the same twig function.

Reply:
The rest LGTM, it works fine

Reply:
@kingjia90 Thank you for your inputs, I made the proposed changes. I have taken the existing Category class and LinkGenerator, this could be a realistic requirement. Also merged the two examples.